deployments/upgrades/export_configs_to_helm.py

```
@@ -91,31 +105,60 @@
+    valid YAML and the operator may apply it without realizing.
+    """
     url = f'{base_url}{path}'
     parsed = urllib.parse.urlsplit(url)
     if parsed.scheme not in ('http', 'https'):
-        print(f'Error: URL must use http or https scheme: {url}',
-              file=sys.stderr)
-        return None
+        raise ExportError(f'URL must use http or https scheme: {url}')
     req = urllib.request.Request(url, headers=headers)
     try:
         with urllib.request.urlopen(req, timeout=30) as resp:
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)